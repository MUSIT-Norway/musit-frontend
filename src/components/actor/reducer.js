import { Observable } from 'rxjs/Rx';
import actions from "./actions";

const dataFromServer = [
  {
    id: 1,
    fn: 'Jarl parl'
  },
  {
    id: 2,
    fn: 'Line pine'
  },
  {
    id: 3,
    fn: 'Kjell pell'
  },
  {
    id: 4,
    fn: 'Stein sein'
  },
  {
    id: 5,
    fn: 'Knut lut'
  },
  {
    id: 6,
    fn: 'Ellen'
  }
]

const initialState = [];

const ActorReducer$ = Observable.of(() => initialState)
  .merge(
    actions.suggest$.map((update) => () => {
      if (!update) {
        return initialState;
      }
      return dataFromServer.filter(d => {
        const name = d.fn.toLowerCase();
        const query = update.value.toLowerCase();
        return name.indexOf(query) > -1
      })
    }),
    actions.clear$.map(() => () => initialState),
  );

export default Observable.merge(
  ActorReducer$.map(reducer => ["suggest", reducer]),
);