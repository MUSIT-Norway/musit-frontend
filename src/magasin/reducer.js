import { toFrontend } from './mapper';
import * as types from './types';

const initialState = {
  root: {
    data: {}
  }
};

const storageUnitGridReducer = (state = initialState, action = {}) => {
  switch (action.type) {
  case types.LOAD_SEVERAL:
    return {
      ...state,
      loading: true
    };
  case types.LOAD_SEVERAL_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: action.result
    };
  case types.LOAD_SEVERAL_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      error: action.error
    };
  case types.LOAD_ONE_SUCCESS:
    return {
      ...state,
      root: {
        ...state.root,
        loading: false,
        loaded: true,
        data: toFrontend(action.result)
      }
    };
  case types.LOAD_ONE_FAIL:
    return {
      ...state,
      root: {
        ...state.root,
        loading: false,
        loaded: false,
        error: action.error
      }
    };
  case types.CLEAR_ROOT: {
    return {
      ...state,
      root: {}
    };
  }
  case types.DELETE:
    return {
      ...state,
      loading: true
    };
  case types.DELETE_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: []
    };
  case types.DELETE_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      error: action.error
    };
  default:
    return state;
  }
};

export default storageUnitGridReducer;