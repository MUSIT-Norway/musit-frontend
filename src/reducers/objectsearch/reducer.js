import { omit } from 'lodash';
import { getPath } from '../helper';
import MusitObject from '../../models/object';
import * as types from './constants';

const initialState = {
  data: {
    totalMatches: 0,
    matches: []
  },
  params: {
    currentPage: 1,
    perPage: 50
  },
  loaded: true
};

export default (state = initialState, action) => {
  switch(action.type) {
  case types.CHANGE_FIELD:
    return {
      ...state,
      params: {
        ...state.params,
        [action.field]: action.value
      }
    };
  case types.SEARCH_OBJECTS:
    return {
      ...omit(state, 'error'),
      loading: true,
      loaded: false,
      data: {
        totalMatches: 0,
        matches: []
      },
      params: {
        ...state.params,
        currentPage: action.page
      }
    };
  case types.SEARCH_OBJECTS_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      data: [],
      error: action.error
    };
  case types.SEARCH_OBJECTS_SUCCESS:
    return {
      ...omit(state, 'error'),
      loading: false,
      loaded: true,
      data: {
        totalMatches: action.result.totalMatches,
        matches: action.result.matches.map(data => {
          return new MusitObject({
            ...data,
            breadcrumb: getPath(data)
          });
        })
      }
    };
  default:
    return state;
  }
};