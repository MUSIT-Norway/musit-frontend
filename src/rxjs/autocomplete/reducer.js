import { Observable } from 'rxjs/Rx';

export default function createReducer(actions) {
  const initialState = [];
  return Observable.of(() => initialState)
    .merge(
      actions.update$.map((data) => () => data),
      actions.clear$.map(() => () => initialState),
    );
}