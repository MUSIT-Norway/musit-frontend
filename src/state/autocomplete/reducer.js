import { Observable } from 'rxjs/Rx';

export default function createReducer(actions, stateKey = "suggest") {
  const initialState = [];
  const reducer$ = Observable.of(() => initialState)
    .merge(
      actions.update$.map((data) => () => data),
      actions.clear$.map(() => () => initialState),
    );
  return Observable.merge(
    reducer$.map(reducer => [stateKey, reducer]),
  );
}