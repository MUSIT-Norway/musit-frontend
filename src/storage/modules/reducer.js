import { toFrontend } from '../../magasin/modules/mapper';
import * as types from './types';

const initialState = {
  state: {
    environmentRequirement: {},
    environmentAssessment: {},
    securityAssessment: {}
  }
};

const storageUnitContainerReducer = (state = initialState, action = {}) => {
  switch (action.type) {
  case types.INSERT:
    return {
      ...state,
      loading: true
    };
  case types.INSERT_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: toFrontend(action.result)
    };
  case types.INSERT_FAIL:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: action.error
    };
  case types.UPDATE:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: action.data
    };
  case types.LOAD:
    return {
      ...state,
      loading: true
    };
  case types.LOAD_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: toFrontend(action.result)
    };
  case types.LOAD_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      error: action.error
    };
  case types.UPDATE_STATE:
    return {
      ...state,
      state: action.data
    };
  case types.CLEAR_STATE:
    return {
      ...state,
      state: {
        environmentRequirement: {},
        environmentAssessment: {},
        securityAssessment: {}
      }
    };
  default:
    return state;
  }
};

export default storageUnitContainerReducer;