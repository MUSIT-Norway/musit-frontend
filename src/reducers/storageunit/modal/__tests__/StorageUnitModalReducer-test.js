import assert from 'assert'
import deepFreeze from 'deep-freeze'

import * as actions from '../index'
import reducer from '../index'
import Config from '../../../../config'
import request from 'superagent';
import nocker from 'superagent-nock';
const nock = nocker(request);

import suReducer, {
  clearRoot
} from '../index'

const testRootState = {
  data: [
    {
      id: 1,
      type: 'Building',
      area: 1,
      name: 'B1',
      areaTo: 1,
      groupRead: 'foo',
      height: 1,
      heightTo: 1,
      address: 'Foo street 1, 1234 Bar',
      groupWrite: 'bar'
    }
  ],
  root: {
    data: {}
  }
}
deepFreeze(testRootState)
const testSubRootState = {
  data: [
    {
      id: 2,
      type: 'Room',
      area: 1,
      name: 'R1',
      areaTo: 1,
      groupRead: 'foo',
      height: 1,
      heightTo: 1,
      groupWrite: 'bar'
    },
    {
      id: 3,
      type: 'Room',
      area: 1,
      name: 'R2',
      areaTo: 1,
      groupRead: 'foo',
      height: 1,
      heightTo: 1,
      groupWrite: 'bar'
    }
  ],
  root: {
    data: {
      id: 1,
      type: 'Building',
      area: 1,
      name: 'B1',
      areaTo: 1,
      groupRead: 'foo',
      height: 1,
      heightTo: 1,
      address: 'Foo street 1, 1234 Bar',
      groupWrite: 'bar'
    }
  }
}
deepFreeze(testSubRootState)
const testSubNoRootState = {
  data: [
    {
      id: 2,
      type: 'Room',
      area: 1,
      name: 'R1',
      areaTo: 1,
      groupRead: 'foo',
      height: 1,
      heightTo: 1,
      groupWrite: 'bar'
    },
    {
      id: 3,
      type: 'Room',
      area: 1,
      name: 'R2',
      areaTo: 1,
      groupRead: 'foo',
      height: 1,
      heightTo: 1,
      groupWrite: 'bar'
    }
  ],
  root: {}
}
deepFreeze(testSubNoRootState)


describe('StorageUnitModalReducer', () => {
  it('Initial state is set', () => {
    const state = suReducer(testRootState, {})
    assert(state === testRootState)
  })

  it('Clear StorageUnitModalReducer root', () => {
    const state = suReducer(testSubRootState, clearRoot())
    assert(JSON.stringify(state) === JSON.stringify(testSubNoRootState))
  })

  const loadOneState = {
    id: 2,
    name: 'Utviklingsmuseet',
    isPartOf: 1,
    path: ',1,2,',
    pathNames: [
      {
        nodeId: 1,
        name: 'root-node'
      },
      {
        nodeId: 2,
        name: 'Utviklingsmuseet'
      }],
    updatedBy: 123,
    updatedDate: '2016-01-01T00:00:00+00:00',
    type: 'Organisation'
  }

  it('creates LOAD_ROOT_SUCCESS when fetching data has been done', () => {
    const id = 2
    const url = `${Config.magasin.urls.storagefacility.baseUrl(1)}/${id}`
    nock('http://localhost')
        .get(url)
        .reply(200, loadOneState)
    const store = mockStore()

    return store.dispatch(actions.loadNode(2))
        .then(() => {
          expect(store.getActions()).toMatchSnapshot()
        })
  })

  it('LOAD_ROOT: no action', () => {
    expect(
        reducer(undefined, {})
    ).toMatchSnapshot()
  })

  it('LOAD_ROOT: initial action', () => {
    expect(
        reducer(undefined, {
          type: actions.LOAD_ROOT
        })
    ).toMatchSnapshot()
  })

  it('LOAD_ROOT: success action', () => {
    expect(
        reducer(undefined, {
          type: actions.LOAD_ROOT_SUCCESS,
          result: {
            someField: 1
          }
        })
    ).toMatchSnapshot()
  })

  it('LOAD_ROOT: fail action', () => {
    expect(
        reducer(undefined, {
          type: actions.LOAD_ROOT_FAIL,
          error: Error('LOAD_ROOT: has error.')
        })
    ).toMatchSnapshot()
  })

  const loadSeveralChildState = [
    {
      id: 3,
      name: 'Forskningens hus',
      isPartOf: 2,
      path: ',1,2,3,',
      type: 'Building',
      updatedBy: 123,
      updatedDate: '2016-01-01T00:00:00+00:00'
    }]
  it('creates LOAD_SEVERAL_SUCCESS when fetching data has been done', () => {
    const url = `${Config.magasin.urls.storagefacility.baseUrl(1)}/1`
    nock('http://localhost')
        .get(url)
        .reply(200, loadSeveralChildState)
    const store = mockStore()

    return store.dispatch(actions.loadNode(1))
        .then(() => {
          expect(store.getActions()).toMatchSnapshot()
        })
  })

  it('LOAD_SEVERAL: no action', () => {
    expect(
        reducer(undefined, undefined)
    ).toMatchSnapshot()
  })

  it('LOAD_SEVERAL: initial action', () => {
    expect(
        reducer(undefined, {
          type: actions.LOAD_SEVERAL
        })
    ).toMatchSnapshot()
  })

  it('LOAD_SEVERAL: success action', () => {
    expect(
        reducer(undefined, {
          type: actions.LOAD_SEVERAL_SUCCESS,
          result: {
            someField: 1
          }
        })
    ).toMatchSnapshot()
  })

  it('LOAD_SEVERAL: fail action', () => {
    expect(
        reducer(undefined, {
          type: actions.LOAD_SEVERAL_FAIL,
          error: Error('LOAD_SEVERAL has error.')
        })
    ).toMatchSnapshot()
  })

  it('creates LOAD_SEVERAL_SUCCESS child when fetching data has been done', () => {
    const id = 2;
    const url = `${Config.magasin.urls.storagefacility.baseUrl(1)}/${id}/children`;
    nock('http://localhost')
        .get(url)
        .reply(200, loadSeveralChildState)
    const store = mockStore()

    return store.dispatch(actions.loadChildren(id))
        .then(() => {
          expect(store.getActions()).toMatchSnapshot()
        })
  })

  it('LOAD_SEVERAL child: no action', () => {
    expect(
        reducer(undefined, undefined)
    ).toMatchSnapshot()
  })

  it('LOAD_SEVERAL child: initial action', () => {
    expect(
        reducer(undefined, {
          type: actions.LOAD_SEVERAL
        })
    ).toMatchSnapshot()
  })

  it('LOAD_SEVERAL child: success action', () => {
    expect(
        reducer(undefined, {
          type: actions.LOAD_SEVERAL_SUCCESS,
          result: {
            someField: 1
          }
        })
    ).toMatchSnapshot()
  })

  it('LOAD_SEVERAL child: fail action', () => {
    expect(
        reducer(undefined, {
          type: actions.LOAD_SEVERAL_FAIL,
          error: Error('LOAD_SEVERAL child has error.')
        })
    ).toMatchSnapshot()
  })


  it('SET_CURRENT action', () => {
    expect(
        reducer(loadSeveralChildState, actions.setCurrent(1))
    ).toMatchSnapshot()
  })

  it('CLEAR_CURRENT action', () => {
    expect(
        reducer(loadSeveralChildState, actions.clearCurrent())
    ).toMatchSnapshot()
  })
})
