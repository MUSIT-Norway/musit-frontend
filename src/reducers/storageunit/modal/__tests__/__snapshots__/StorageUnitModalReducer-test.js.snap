exports[`StorageUnitModalReducer CLEAR_CURRENT action 1`] = `
Object {
  "0": Object {
    "id": 3,
    "isPartOf": 2,
    "name": "Forskningens hus",
    "path": ",1,2,3,",
    "type": "Building",
    "updatedBy": 123,
    "updatedDate": "2016-01-01T00:00:00+00:00"
  },
  "currentId": null
}
`;

exports[`StorageUnitModalReducer LOAD_ROOT: fail action 1`] = `
Object {
  "root": Object {
    "error": [Error: LOAD_ROOT: has error.],
    "loaded": false,
    "loading": false
  }
}
`;

exports[`StorageUnitModalReducer LOAD_ROOT: initial action 1`] = `
Object {
  "loading": true,
  "root": Object {}
}
`;

exports[`StorageUnitModalReducer LOAD_ROOT: no action 1`] = `
Object {
  "root": Object {}
}
`;

exports[`StorageUnitModalReducer LOAD_ROOT: success action 1`] = `
Object {
  "root": Object {
    "data": Object {
      "path": Array [],
      "someField": 1
    },
    "loaded": true,
    "loading": false
  }
}
`;

exports[`StorageUnitModalReducer LOAD_SEVERAL child: fail action 1`] = `
Object {
  "error": [Error: LOAD_SEVERAL child has error.],
  "loaded": false,
  "loading": false,
  "root": Object {}
}
`;

exports[`StorageUnitModalReducer LOAD_SEVERAL child: initial action 1`] = `
Object {
  "loading": true,
  "root": Object {}
}
`;

exports[`StorageUnitModalReducer LOAD_SEVERAL child: no action 1`] = `
Object {
  "root": Object {}
}
`;

exports[`StorageUnitModalReducer LOAD_SEVERAL child: success action 1`] = `
Object {
  "data": Object {
    "someField": 1
  },
  "loaded": true,
  "loading": false,
  "root": Object {}
}
`;

exports[`StorageUnitModalReducer LOAD_SEVERAL: fail action 1`] = `
Object {
  "error": [Error: LOAD_SEVERAL has error.],
  "loaded": false,
  "loading": false,
  "root": Object {}
}
`;

exports[`StorageUnitModalReducer LOAD_SEVERAL: initial action 1`] = `
Object {
  "loading": true,
  "root": Object {}
}
`;

exports[`StorageUnitModalReducer LOAD_SEVERAL: no action 1`] = `
Object {
  "root": Object {}
}
`;

exports[`StorageUnitModalReducer LOAD_SEVERAL: success action 1`] = `
Object {
  "data": Object {
    "someField": 1
  },
  "loaded": true,
  "loading": false,
  "root": Object {}
}
`;

exports[`StorageUnitModalReducer SET_CURRENT action 1`] = `
Object {
  "0": Object {
    "id": 3,
    "isPartOf": 2,
    "name": "Forskningens hus",
    "path": ",1,2,3,",
    "type": "Building",
    "updatedBy": 123,
    "updatedDate": "2016-01-01T00:00:00+00:00"
  },
  "currentId": 1
}
`;

exports[`StorageUnitModalReducer creates LOAD_ROOT_SUCCESS when fetching data has been done 1`] = `
Array [
  Object {
    "type": "musit/storageunit-modal/LOAD_ROOT"
  },
  Object {
    "result": Object {
      "id": 2,
      "isPartOf": 1,
      "name": "Utviklingsmuseet",
      "path": ",1,2,",
      "pathNames": Array [
        Object {
          "name": "root-node",
          "nodeId": 1
        },
        Object {
          "name": "Utviklingsmuseet",
          "nodeId": 2
        }
      ],
      "type": "Organisation",
      "updatedBy": 123,
      "updatedDate": "2016-01-01T00:00:00+00:00"
    },
    "type": "musit/storageunit-modal/LOAD_ROOT_SUCCESS"
  }
]
`;

exports[`StorageUnitModalReducer creates LOAD_SEVERAL_SUCCESS child when fetching data has been done 1`] = `
Array [
  Object {
    "type": "musit/storageunit-modal/LOAD_SEVERAL"
  },
  Object {
    "result": Array [
      Object {
        "id": 3,
        "isPartOf": 2,
        "name": "Forskningens hus",
        "path": ",1,2,3,",
        "type": "Building",
        "updatedBy": 123,
        "updatedDate": "2016-01-01T00:00:00+00:00"
      }
    ],
    "type": "musit/storageunit-modal/LOAD_SEVERAL_SUCCESS"
  }
]
`;

exports[`StorageUnitModalReducer creates LOAD_SEVERAL_SUCCESS when fetching data has been done 1`] = `
Array [
  Object {
    "type": "musit/storageunit-modal/LOAD_ROOT"
  },
  Object {
    "result": Array [
      Object {
        "id": 3,
        "isPartOf": 2,
        "name": "Forskningens hus",
        "path": ",1,2,3,",
        "type": "Building",
        "updatedBy": 123,
        "updatedDate": "2016-01-01T00:00:00+00:00"
      }
    ],
    "type": "musit/storageunit-modal/LOAD_ROOT_SUCCESS"
  }
]
`;
